examples/app-with-on-behalf-of-auth/pkg/plugin/resources.go

```diff
@@ -0,0 +1,112 @@
+package plugin
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/go-jose/go-jose/v3/jwt"
+)
+
+// parseToken parses the given JWT token and returns the headers and claims.
+func (a *App) parseToken(token string) (map[string]interface{}, error) {
+	res := map[string]interface{}{}
+	parsedJWT, err := jwt.ParseSigned(token)
+	if err != nil {
+		return nil, err
+	}
+	claims := map[string]interface{}{}
+	err = parsedJWT.UnsafeClaimsWithoutVerification(&claims)
+	if err != nil {
+		return nil, err
+	}
+	res["headers"] = parsedJWT.Headers
+	res["claims"] = claims
+	return res, nil
+}
+
+// handleAPI handles requests to the /api endpoint.
+func (a *App) handleAPI(w http.ResponseWriter, req *http.Request) {
+	proxyMethod := req.FormValue("method")
+	if proxyMethod == "" {
+		proxyMethod = "GET"
+	}
+	proxyBody := req.FormValue("body")
+	var bodyReader io.Reader
+	if proxyBody != "" {
+		bodyReader = bytes.NewReader([]byte(proxyBody))
+	}
+
+	proxyReq, err := http.NewRequest(proxyMethod, a.grafanaAppURL+req.URL.Path, bodyReader)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	userID := req.FormValue("userID")
+	var token string
+	if userID == "" {
+		token, err = a.tokenRetriever.Self(req.Context())
+	} else {
+		token, err = a.tokenRetriever.OnBehalfOfUser(req.Context(), userID)
+	}
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	proxyReq.Header.Set("Authorization", "Bearer "+token)
+
+	// TODO: Make this configurable.
```

Comment:
What do you mean by this? Just to read the org id from PluginContext?

Reply:
That's a good question. The `X-Grafana-Org-Id` header is something new, @mgyongyosi added it to make it work (it was not part of the original version of the plugin). What's the goal here @mgyongyosi ? Can this header be set automatically based on the Org-Id of the request (I am not sure if that value is set when using the `CallResourceHandler`) or is it okay to hardcode it to some value?

Reply:
The PoC implementation of the single-tenant external service auth only supports the default organization (orgId = 1), so for now it's okay to be harcoded to `1`.

Reply:
:+1: added that as a comment